### PR TITLE
0.9.7

### PR DESCRIPTION
**Pull request by smaeda-ks**
This PR includes:

- bump minimum supported Ruby version to 2.5
https://www.ruby-lang.org/en/news/2020/04/05/support-of-ruby-2-4-has-ended/
- remove `--no-ssl` option as all endpoints are now forced to use TLS anyway
- workaround for OpenStruct bug in Ruby 3.0.0 https://github.com/twitter/twurl/issues/159#issuecomment-762511975 https://github.com/twitter/twurl/issues/160
- `-v` option to include Ruby runtime version
- fix escaping issues with `-d` option https://github.com/twitter/twurl/issues/157
- changing `--raw-data` option behavior

#### changing `--raw-data` option behavior

Although the description says `Sends the specified data as it is...` it's actually parsing input data using `CGI.parse()` method. This isn't ideal, especially when it comes to JSON data, for instance. We shouldn't touch the specified raw data at all.

This commit changes this behavior and makes this option a real thing, so users can also send JSON data using `-r` option by properly setting the `content-type` request header using `-A` option. For backword-compatibility, we still allow `-d` option to send JSON data as long as `content-type: application/json` request header is set along with it.

Also, since `-r` takes input data as-is, it doesn't make sense to allow combine use with `-d` option or adding `-r` option more than once, so we treat them as an error going forward.

> ERROR: can't specify '-r' option more than once
> ERROR: can't use '-r' and '-d' options together

### CODE REVIEW

**Comment by smaeda-ks** [comment]
Hi @andypiper, let me know if you have any questions. Not sure why Travis CI is stuck with "received" status but once it's started this should pass tests on all versions. Thanks

**Comment by smaeda-ks** [comment]
https://travis-ci.com/github/twitter/twurl/builds/213791826 all green :)

**Comment by smaeda-ks** [comment]
Also, since we're not really using Coveralls, can just remove it from `test_helper.rb`:

```diff
- SimpleCov.formatters = [SimpleCov::Formatter::HTMLFormatter, Coveralls::SimpleCov::Formatter]
+ SimpleCov.formatters = [SimpleCov::Formatter::HTMLFormatter]
```

https://github.com/lemurheavy/coveralls-ruby/issues/161

File: lib/twurl/app_only_oauth_client.rb [hunk 1]
```diff
@@ -57,12 +57,9 @@ def set_http_client_options(http)
       http.set_debug_output(Twurl.options.debug_output_io) if Twurl.options.trace
       http.read_timeout = http.open_timeout = Twurl.options.timeout || 60
       http.open_timeout = Twurl.options.connection_timeout if Twurl.options.connection_timeout
-      # Only override if Net::HTTP support max_retries (since Ruby >= 2.5)
-      http.max_retries = 0 if http.respond_to?(:max_retries=)
-      if Twurl.options.ssl?
```
Comment (osowskit) [inline lib/twurl/app_only_oauth_client.rb:62]: was removing the SSL option to always turn off SSL verification intentional?

Comment (andypiper) [reply lib/twurl/app_only_oauth_client.rb:62]: I believe so - the API requires HTTPS / TLS connections.

**Comment by smaeda-ks** [comment]
No activities for code review/approval for more than a year. Closing this PR.

**Comment by andypiper** [comment]
@smaeda-ks review is happening now - thank you so much for this. Re-opening.

**Comment by smaeda-ks** [comment]
Hi @andypiper, this PR should be ready for merging. Added a few comments due to https://github.com/twitter/twurl/pull/173 but should be all set.

Could you please merge this so we can release 0.9.7? Thanks

**Pull request merged**